telcell/celldb/pgdatabase.py

```diff
@@ -138,11 +141,20 @@ def search(
         count_limit: Optional[int] = 10000,
         random_order: bool = False,
         exclude: Optional[List[CellIdentity]] = None,
+        ci: Optional[CellIdentity] = None,
     ) -> CellCollection:
         qwhere = list(self._qwhere)
         qargs = list(self._qargs)
 
-        if coords is not None and distance_limit_m is not None:
+        if coords is not None and distance_limit_m is None and count_limit is None:
```

Comment:
Are the `None`s here strictly needed? This reads more like English when treated as truthiness imo:

```suggestion
        if coords and not (distance_limit_m or count_limit):
```